go/libraries/doltcore/sqle/expranalysis/expranalysis.go

```diff
@@ -84,6 +84,12 @@ func stripTableNamesFromExpression(expr sql.Expression, quoted bool) sql.Express
 	return e
 }
 
+// Session is a way to access the Dolt session without import cycles.
+type Session interface {
```

Comment:
(minor) Any other ideas for names for this? Seems okay, and I get what you're trying to do here, but feels a little too generic.